pkg/apps/store/store.go

```diff
@@ -147,39 +149,68 @@ func getCR(cr interface{}) (*customResource, error) {
 	return &customResource{namespace: namespace, size: size, imageRegistries: imageRegistries, registryConfiguration: &registryConfiguration}, nil
 }
 
-func generateRc(v types.PublicVersion, config *protoform.Config, kubeclient *kubernetes.Clientset, customResource *customResource, cr interface{}) ([]*components.ReplicationController, error) {
+func generateRc(v types.PublicVersion, config *protoform.Config, kubeclient *kubernetes.Clientset, sizeClient *sizeclientset.Clientset, customResource *customResource, cr interface{}) ([]*components.ReplicationController, error) {
 	rcs := make([]*components.ReplicationController, 0)
 	// Size
-	sizegen, ok := ComponentStore.Size[v.Size]
-	if !ok {
-		return nil, fmt.Errorf("size %s couldn't be found", v.Size)
+
+	//sizegen, ok := ComponentStore.Size[v.Size]
+	//if !ok {
+	//	return nil, fmt.Errorf("size %s couldn't be found", v.Size)
+	//}
+	//
+	//componentSize := sizegen.GetSize(customResource.size)
+	//if componentSize == nil {
+	//	return nil, fmt.Errorf("size %s couldn't be found in %s", customResource.size, v.Size)
+	//}
+
+	sizeName := getSizeName(cr)
```

Comment:
customResource.size will already have the size info. `getSizeName` method can be removed

Reply:
done